pyteal/ast/frame.py

```diff
@@ -1,16 +1,18 @@
 from typing import TYPE_CHECKING
 
 from pyteal.ast.expr import Expr
-from pyteal.types import TealType, require_type
-from pyteal.errors import TealInputError, verifyProgramVersion
+from pyteal.ast.int import Int
+from pyteal.ast.abstractvar import AbstractVar
+from pyteal.types import TealType, require_type, types_match
+from pyteal.errors import TealInputError, TealInternalError, verifyProgramVersion
 from pyteal.ir import TealBlock, TealSimpleBlock, TealOp, Op
 
 if TYPE_CHECKING:
     from pyteal.compiler import CompileOptions
 
 
 class Proto(Expr):
-    def __init__(self, num_args: int, num_returns: int):
+    def __init__(self, num_args: int, num_returns: int, /, *, num_allocs: int = 0):
```

Comment:
```suggestion
    def __init__(self, num_args: int, num_returns: int, *, num_allocs: int = 0):
```

In my opinion, it is convenient to continue allowing the naming of arguments `num_args` and `num_returns`. However, there could be a compelling reason to disallow this so feel free to ignore this suggestion.

Reply:
the interface changed by introduction of #593 , resolving